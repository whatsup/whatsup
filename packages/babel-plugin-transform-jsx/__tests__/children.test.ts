import { transform } from './__config__/transform'

describe('transform children', function () {
    it('Should transform without children', function () {
        expect(transform('<div></div>')).toMatchSnapshot()
    })

    it('Should transform once child', function () {
        expect(transform('<div><span /></div>')).toMatchSnapshot()
    })

    it('Should transform many children', function () {
        expect(transform('<ul><li /><li /></ul>')).toMatchSnapshot()
    })

    it('Should transform children & props', function () {
        expect(transform('<ul prop="1" prop2={3}><li /><li /></ul>')).toMatchSnapshot()
    })

    it('Should transform children with primitives', function () {
        expect(transform('<div><span>foo</span>bar<i>baz</i></div>')).toMatchSnapshot()
    })

    it('Should transform children spread', function () {
        expect(transform('<div><span>foo</span>{...other}</div>')).toMatchSnapshot()
    })

    it('Should transform children with expressions', function () {
        expect(transform('<div><span>foo</span>{[`one`, `two`].map(i => <span>{i}</span>)}</div>')).toMatchSnapshot()
    })

    it('Should trim whitespaces right only', function () {
        expect(transform('<div>{count} items </div>')).toMatchSnapshot()
    })

    it('Should trim whitespaces left only', function () {
        expect(transform('<div> items {count}</div>')).toMatchSnapshot()
    })

    it('Should trim whitespaces and create empty children', function () {
        expect(transform('<div>    </div>')).toMatchSnapshot()
    })

    it('Should safe whitespaces between children', function () {
        expect(transform('<div>User {name} {age}</div>')).toMatchSnapshot()
    })

    it('Should trim newlines ', function () {
        expect(
            transform(`<div>
            <div></div>
            <div></div>
        </div>`)
        ).toMatchSnapshot()
    })

    it('Should trim newlines 2', function () {
        expect(
            transform(`<div>
            <div>child1</div>
            <div>child2</div>
            child3
            {1612}
            {null}
            {false}
            {true}
        </div>`)
        ).toMatchSnapshot()
    })
})
