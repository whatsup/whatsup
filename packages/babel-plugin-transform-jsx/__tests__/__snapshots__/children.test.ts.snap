// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform children Should safe whitespaces between children 1`] = `
"[\\"<0~UniqueId>\\", \\"div\\", {
  children: [\\"User \\", name, \\" \\", age]
}, undefined];"
`;

exports[`transform children Should transform children & props 1`] = `
"[\\"<0~UniqueId>\\", \\"ul\\", {
  children: [[\\"<0~UniqueId>\\", \\"li\\", {}, undefined], [\\"<0~UniqueId>\\", \\"li\\", {}, undefined]],
  prop: \\"1\\",
  prop2: 3
}, undefined];"
`;

exports[`transform children Should transform children spread 1`] = `
"[\\"<0~UniqueId>\\", \\"div\\", {
  children: [[\\"<0~UniqueId>\\", \\"span\\", {
    children: \\"foo\\"
  }, undefined], ...other]
}, undefined];"
`;

exports[`transform children Should transform children with expressions 1`] = `
"[\\"<0~UniqueId>\\", \\"div\\", {
  children: [[\\"<0~UniqueId>\\", \\"span\\", {
    children: \\"foo\\"
  }, undefined], [\`one\`, \`two\`].map(i => [\\"<0~UniqueId>\\", \\"span\\", {
    children: i
  }, undefined])]
}, undefined];"
`;

exports[`transform children Should transform children with primitives 1`] = `
"[\\"<0~UniqueId>\\", \\"div\\", {
  children: [[\\"<0~UniqueId>\\", \\"span\\", {
    children: \\"foo\\"
  }, undefined], \\"bar\\", [\\"<0~UniqueId>\\", \\"i\\", {
    children: \\"baz\\"
  }, undefined]]
}, undefined];"
`;

exports[`transform children Should transform many children 1`] = `
"[\\"<0~UniqueId>\\", \\"ul\\", {
  children: [[\\"<0~UniqueId>\\", \\"li\\", {}, undefined], [\\"<0~UniqueId>\\", \\"li\\", {}, undefined]]
}, undefined];"
`;

exports[`transform children Should transform once child 1`] = `
"[\\"<0~UniqueId>\\", \\"div\\", {
  children: [\\"<0~UniqueId>\\", \\"span\\", {}, undefined]
}, undefined];"
`;

exports[`transform children Should transform without children 1`] = `"[\\"<0~UniqueId>\\", \\"div\\", {}, undefined];"`;

exports[`transform children Should trim newlines  1`] = `
"[\\"<0~UniqueId>\\", \\"div\\", {
  children: [[\\"<0~UniqueId>\\", \\"div\\", {}, undefined], [\\"<0~UniqueId>\\", \\"div\\", {}, undefined]]
}, undefined];"
`;

exports[`transform children Should trim newlines 2 1`] = `
"[\\"<0~UniqueId>\\", \\"div\\", {
  children: [[\\"<0~UniqueId>\\", \\"div\\", {
    children: \\"child1\\"
  }, undefined], [\\"<0~UniqueId>\\", \\"div\\", {
    children: \\"child2\\"
  }, undefined], \\" child3 \\", 1612, null, false, true]
}, undefined];"
`;

exports[`transform children Should trim whitespaces and create empty children 1`] = `"[\\"<0~UniqueId>\\", \\"div\\", {}, undefined];"`;

exports[`transform children Should trim whitespaces left only 1`] = `
"[\\"<0~UniqueId>\\", \\"div\\", {
  children: [\\"items \\", count]
}, undefined];"
`;

exports[`transform children Should trim whitespaces right only 1`] = `
"[\\"<0~UniqueId>\\", \\"div\\", {
  children: [count, \\" items\\"]
}, undefined];"
`;
